pkg/workspace/image/pusher.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/sh
+set -ex
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+TMPDIR="$(mktemp -d)"
+
+VOL_DIR="${1}"
+[ ! -z "${VOL_DIR}" ] || VOL_DIR='{{ .volDir }}'
+
+IMG_REF="${2}"
+[ ! -z "${IMG_REF}" ] || IMG_REF='{{ .imgRef }}'
+
+#{{`
+
+wait() {
+    until [ -e "${VOL_DIR}/fine_tuning_completed.txt" ]
+    do
+        sleep 1
+    done
+}
+
+mklayer() {
+    local DATA_DIR="${TMPDIR}/data"
+    mkdir -p "${DATA_DIR}"
+
+    cp -R "${VOL_DIR}/adapter_config.json" "${VOL_DIR}/adapter_model.safetensors" "${DATA_DIR}"
+
+    local TAR_LAYER_PATH="${TMPDIR}/layer.tar"
+
+    tar c -f "${TAR_LAYER_PATH}" -C "$(dirname "${DATA_DIR}")" "$(basename "${DATA_DIR}")"
+
+    TAR_LAYER_DIFF="$(sha256sum "${TAR_LAYER_PATH}" | cut -d ' ' -f '1')"
+
+    gzip -9 "${TAR_LAYER_PATH}"
+
+    TGZ_LAYER_MIME='application/vnd.oci.image.layer.v1.tar+gzip'
+    TGZ_LAYER_PATH="${TAR_LAYER_PATH}.gz"
+}
+
+mkconfig() {
+    CONFIG_MIME='application/vnd.oci.image.config.v1+json'
+    CONFIG_PATH="${TMPDIR}/config.json"
+
+    printf '{"rootfs":{"diff_ids":["sha256:%s"]}}' "${TAR_LAYER_DIFF}" > "${CONFIG_PATH}"
+}
+
+mklayout() {
+    LAYOUT_REF="${TMPDIR}/layout:latest"
+
+    cd "$(dirname "${TGZ_LAYER_PATH}")"
+    oras push --disable-path-validation --config "${CONFIG_PATH}:${CONFIG_MIME}" --oci-layout "${LAYOUT_REF}" "$(basename "${TGZ_LAYER_PATH}"):${TGZ_LAYER_MIME}"
```

Comment:
Do oras add some metadata to the image manifest? e.g. creation timestamp.
We can add some metadata to this image to make it easier to maintain.
e.g. maintainer: kaito.dev.xxx;   target model name this adapter files can be used for.

Reply:
By default, it only sets `org.opencontainers.image.created`.

![image](https://github.com/user-attachments/assets/82deb2fa-0b98-4838-98a8-282c90f68bfb)

Reply:
We can add more stuff if we would like

Reply:
Can you get the model's name in controller code simply? If so, we can add a model name annotation here.
This is the most useful information for this image.

Reply:
What do you think the label key should be? `sh.kaito.model`? `sh.kaito.model.name`? `sh.kaito.model-name`?

Reply:
`sh.kaito.model.name` this one works for me.

Reply:
Done!